python/lsst/afw/table/_base.py

```diff
@@ -115,6 +115,13 @@ def __getitem__(self, key):
         elif isinstance(key, np.ndarray):
             if key.dtype == bool:
                 return self.subset(key)
+            if np.issubdtype(key.dtype, np.integer) and key.ndim == 1:
+                # Return a shallow copy containing the desired records
+                new = type(self)(self.schema)
+                new.reserve(len(key))
```

Comment:
I don't think we want this for a shallow copy - that will both reserve space in the `vector`-of-`shared_ptr` (_maybe_ what we want, _if_ we're willing to assume it's unlikely that this will be followed by more appends, but the performance without it is guaranteed to scale fine anyway) _and_ allocate new backing-store space for new _deep_-copied records (not what we want at all).